server/controllers/stock/index.js

```diff
@@ -530,12 +531,19 @@ function listLotsDepot(req, res, next) {
     params.defaultPeriodEntry = params.defaultPeriod;
     delete params.defaultPeriod;
   }
-  core.getLotsDepot(null, params)
-    .then((rows) => {
-      res.status(200).json(rows);
-    })
-    .catch(next)
-    .done();
+  try {
```

Comment:
So, if I am reading this code correctly, if we have N lots in the depot, we will issue N + 1 SQL requests:
```js
// first SQL request (num SQL requests = 1)
const data = await core.getLotsDepot(null, params);


const rows = await Promise.all(data.map(async (line) => {
  // one SQL request for N tags 
  const tags = await Lots.getLotTags(db.bid(line.uuid));
  line.tags = tags;
  return line;
}));

// Total SQL requests N + 1
```

I'm not sure something like a raspberry pi could handle this kind of a query.  Especially if no one uses tags, we'll be slowing down our queries for nothing.

I propose you either:
  1. Do something like `GROUP_CONCAT()` to get the tag information in a single SQL query.  This will be messy, but doable.
  2. Do two queries and stitch the results together in javascript.  You are already looping through each lot once, and it will save us from hammering our SQL server.  It would look something like this:
```js
const lotUuids = data.map(row => db.bid(row.uuid));
const tags = await db.exec('SELECT lot_uuid, tag.color /*, etc ... */ WHERE lt.lot_uuid IN (?)', [lotUuids]);


// make a lot_uuid -> tags map.
const tagMap = _.groupdBy(tags, 'lot_uuid');

data.forEach(lot => {
  lot.tags = tagMap[lot.uuid];
});

// Total SQL requests 2.
// Loops are O(N + M) where length(lots) is N and length(tags) is M.

```

Reply:
I suppose as an improvement, you can check if `tags.length === 0` and return early if that is the case.  It would save a for loop in the default case.